encoder_test.go

```diff
@@ -74,11 +75,23 @@ func TestTextDecoder(t *testing.T) {
 		{"Array(UInt32)", "[]", []uint32{}},
 	}
 
-	dec := &textDecoder{location: time.UTC}
-	for _, tc := range testCases {
+	dec := &textDecoder{location: time.UTC, useDBLocation: false}
+	for i, tc := range testCases {
 		v, err := dec.Decode(tc.tt, []byte(tc.value))
-		if assert.NoError(t, err) {
+		if assert.NoError(t, err, "%d", i) {
 			assert.Equal(t, tc.expected, v)
 		}
 	}
 }
+
+func TestDecodeTimeWithLocation(t *testing.T) {
+	dt := time.Date(2011, 3, 6, 3, 20, 0, 0, time.UTC)
+	dataType := "DateTime(\\'Europe/Moscow\\')"
```

Comment:
Which are different from test case in line 67 on this file ?

Reply:
In test case on line 67 database location is discarded (`useLocation: false` on line 78). In this test location is taken into account and time is converted from Europe\Moscow to UTC.

Reply:
ok